src/features/cosmos/components/Modals/TxFeeRadioGroup.tsx

```diff
@@ -0,0 +1,130 @@
+import { Box, Button, ButtonGroup, Radio, Spinner, useColorModeValue } from '@chakra-ui/react'
+import { chainAdapters } from '@shapeshiftoss/types'
+import { useController, useFormContext, useWatch } from 'react-hook-form'
+import { Amount } from 'components/Amount/Amount'
+import { Text } from 'components/Text'
+
+import { SendFormFields, SendInput } from './Form'
+import { FeePrice } from './views/Confirm'
+
+type TxFeeRadioGroupProps = {
+  fees: FeePrice | null
+}
+
+function getFeeColor(key: chainAdapters.FeeDataKey): string {
+  switch (key) {
+    case chainAdapters.FeeDataKey.Slow:
+      return 'yellow'
+    case chainAdapters.FeeDataKey.Fast:
+      return 'green'
+    case chainAdapters.FeeDataKey.Average:
+    default:
+      return 'blue'
+  }
+}
+
+function getFeeTranslation(key: chainAdapters.FeeDataKey): string {
+  switch (key) {
+    case chainAdapters.FeeDataKey.Slow:
+      return 'modals.send.sendForm.slow'
+    case chainAdapters.FeeDataKey.Fast:
+      return 'modals.send.sendForm.fast'
+    case chainAdapters.FeeDataKey.Average:
+    default:
+      return 'modals.send.sendForm.average'
+  }
+}
+
+const feesOrder: chainAdapters.FeeDataKey[] = [
+  chainAdapters.FeeDataKey.Slow,
+  chainAdapters.FeeDataKey.Average,
+  chainAdapters.FeeDataKey.Fast
+]
+
+export const TxFeeRadioGroup = ({ fees }: TxFeeRadioGroupProps) => {
+  const { control } = useFormContext<SendInput>()
+  const { field } = useController({
+    name: SendFormFields.FeeType,
+    control,
+    rules: { required: true },
+    defaultValue: chainAdapters.FeeDataKey.Average
+  })
+  const asset = useWatch<SendInput, SendFormFields.Asset>({ name: SendFormFields.Asset })
+  const activeFee = useWatch<SendInput, SendFormFields.FeeType>({ name: SendFormFields.FeeType })
+  const bg = useColorModeValue('gray.50', 'gray.850')
+  const borderColor = useColorModeValue('gray.100', 'gray.750')
+
+  if (!fees) {
+    return (
+      <Box
+        display='flex'
+        flexDir='column'
+        alignItems='center'
+        justifyContent='center'
+        py={2}
+        width='full'
+        height='auto'
+      >
```

Comment:
I would use a `Flex` element instead of a box, the code would probably be clearer as this is the role of `Flex`